AK/RedBlackTree.h

```diff
@@ -162,54 +140,54 @@ class BaseRedBlackTree {
             m_minimum = node;
             return;
         } else if (node->key < parent->key) { // we are the left child
-            parent->left_child = node;
+            parent->left_child() = node;
         } else { // we are the right child
-            parent->right_child = node;
+            parent->right_child() = node;
         }
-        node->parent = parent;
+        node->parent() = parent;
 
-        if (node->parent->parent) // no fixups to be done for a height <= 2 tree
+        if (node->parent()->parent()) // no fixups to be done for a height <= 2 tree
```

Comment:
Shouldn't this be `node->parent().parent()` since `Node::parent()` returns a reference?

Reply:
@emanuele6 It is a reference to a pointer. The ugliness of pointers. Is there a good way to make this more readable?

Reply:
> It is a reference to a pointer

Ah, I didn't notice that was a `AK::Array` of `Node*` (makes sense).

I think it's readable enough. I was just confused by the `auto&` return type.